apps/yuudachi/src/functions/locks/locks.ts

```diff
@@ -0,0 +1,47 @@
+import { container, kRedis } from "@yuudachi/framework";
+import type { LocaleParam } from "@yuudachi/framework/types";
+import type { GuildMember } from "discord.js";
+import i18next from "i18next";
+import type { Redis } from "ioredis";
+import { Counter } from "prom-client";
+import { MEMBER_LOCK_EXPIRE_SECONDS } from "../../Constants.js";
+
+const lockCounter = new Counter({
+	name: "yuudachi_bot_v3_utils_locks_acquire_member_lock_total",
+	help: "Total locks acquired",
+	labelNames: ["memberId", "guildId"],
+});
+
+export async function acquireMemberLock(member: GuildMember, locale: LocaleParam, override = false): Promise<void> {
+	const redis = container.resolve<Redis>(kRedis);
+
+	const lockKey = `member-lock:${member.guild.id}:${member.id}`;
+
+	const lock = await redis.set(lockKey, member.user.createdTimestamp, "EX", MEMBER_LOCK_EXPIRE_SECONDS, "NX");
+
+	if (!lock && !override) {
+		throw new Error(
+			i18next.t("command.common.errors.member_lock_acquired", {
+				lng: locale,
+			}),
+		);
+	}
+
+	lockCounter.inc({ memberId: member.id, guildId: member.guild.id });
```

Comment:
I'm not entirely sure what value this provides. However, I _would_ be super interested in a counter tracking expired locks. If a lock expires, that's generally a bad sign and something that should be addressed.

Reply:
Will work on something!! Thanks!

Reply:
Implemented in 9d82acd